From 29222d35c20f961b2992086b28ad4d7d8c7275ed Mon Sep 17 00:00:00 2001
From: Ed Halferty <edhalferty@gmail.com>
Date: Tue, 3 Jul 2018 07:36:05 -0700
Subject: [PATCH 1/2] Adds PHY management with hotplug support for lwip with
 emaclite.

Adds PHY management with hotplug support for lwip with emaclite to resolve BSP compilation issue.

Signed-off-by: Ed Halferty <me@edwardhalferty.com>
Acked-by: Harini Katakam <harinik.katakam@xilinx.com>
---
 .../src/contrib/ports/xilinx/netif/xadapter.c      | 33 ++++++++++++++++++++++
 1 file changed, 33 insertions(+)

diff --git a/ThirdParty/sw_services/lwip202/src/contrib/ports/xilinx/netif/xadapter.c b/ThirdParty/sw_services/lwip202/src/contrib/ports/xilinx/netif/xadapter.c
index 0804158..5bcd5eb 100644
--- a/ThirdParty/sw_services/lwip202/src/contrib/ports/xilinx/netif/xadapter.c
+++ b/ThirdParty/sw_services/lwip202/src/contrib/ports/xilinx/netif/xadapter.c
@@ -304,6 +304,21 @@ static u32_t phy_link_detect(XAxiEthernet *xemacp, u32_t phy_addr)
 	return 0;
 }
 #endif /* XLWIP_CONFIG_INCLUDE_XXVETHERNET */
+#elif defined(XLWIP_CONFIG_INCLUDE_EMACLITE)
+static u32_t phy_link_detect(XEmacLite *xemacp, u32_t phy_addr)
+{
+	u16_t status;
+
+	/* Read Phy Status register twice to get the confirmation of the current
+	 * link status.
+	 */
+	XEmacLite_PhyRead(xemacp, phy_addr, IEEE_STATUS_REG_OFFSET, &status);
+	XEmacLite_PhyRead(xemacp, phy_addr, IEEE_STATUS_REG_OFFSET, &status);
+
+	if (status & IEEE_STAT_LINK_STATUS)
+		return 1;
+	return 0;
+}
 #endif
 
 #if defined(XLWIP_CONFIG_INCLUDE_GEM)
@@ -353,6 +368,21 @@ static u32_t phy_autoneg_status(XAxiEthernet *xemacp, u32_t phy_addr)
 	return 0;
 }
 #endif /* XLWIP_CONFIG_INCLUDE_XXVETHERNET */
+#elif defined(XLWIP_CONFIG_INCLUDE_EMACLITE)
+static u32_t phy_autoneg_status(XEmacLite *xemacp, u32_t phy_addr)
+{
+	u16_t status;
+
+	/* Read Phy Status register twice to get the confirmation of the current
+	 * link status.
+	 */
+	XEmacLite_PhyRead(xemacp, phy_addr, IEEE_STATUS_REG_OFFSET, &status);
+	XEmacLite_PhyRead(xemacp, phy_addr, IEEE_STATUS_REG_OFFSET, &status);
+
+	if (status & IEEE_STAT_AUTONEGOTIATE_COMPLETE)
+		return 1;
+	return 0;
+}
 #endif
 
 void eth_link_detect(struct netif *netif)
@@ -370,6 +400,9 @@ void eth_link_detect(struct netif *netif)
 #else
 	XAxiEthernet *xemacp = &xemacs->axi_ethernet;
 #endif
+#elif defined(XLWIP_CONFIG_INCLUDE_EMACLITE)
+	xemacliteif_s *xemacs = (xemacliteif_s *)(xemac->state);
+	XEmacLite *xemacp = xemacs->instance;
 #endif
 
 	if ((xemacp->IsReady != (u32)XIL_COMPONENT_IS_READY) ||
-- 
2.7.4

